### PR TITLE
New package: wabt

### PR DESCRIPTION
**Pull request by ArchGuyWu**
- The WebAssembly Binary Toolkit is a suite of tools for WebAssembly
- It has a wasm-strip tool, which can be used to strip objects for wasm.

### CODE REVIEW

File: mingw-w64-wabt/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+mingw_arch=('mingw64' 'ucrt64' 'clang64' 'clangarm64')
+url="https://github.com/WebAssembly/wabt"
+msys2_references=(
+  'archlinux: wabt'
+)
+license=('APACHE-2.0')
+depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs")
+makedepends=(
+  "${MINGW_PACKAGE_PREFIX}-cmake"
+  "${MINGW_PACKAGE_PREFIX}-make"
```
Comment (ognevny) [inline mingw-w64-wabt/PKGBUILD:17]:
```suggestion
  "${MINGW_PACKAGE_PREFIX}-ninja"
```

File: mingw-w64-wabt/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  git config submodule."external/ply".url "${srcdir}/ply"
+  git submodule update --recursive
+
+  mkdir build
+}
+
+build() {
+  cd "${_realname}/build"
+
+  export CFLAGS+=" -ffat-lto-objects"
```
Comment (ognevny) [inline mingw-w64-wabt/PKGBUILD:48]: lto is broken in MinGW

File: mingw-w64-wabt/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  cmake \
+    -DCMAKE_INSTALL_PREFIX=${MINGW_PREFIX} \
+    -DCMAKE_BUILD_TYPE=Release \
+    -G "MinGW Makefiles" \
```
Comment (ognevny) [inline mingw-w64-wabt/PKGBUILD:52]:
```suggestion
    -G "Ninja" \
```

File: mingw-w64-wabt/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    ..
+
+  mingw32-make
```
Comment (ognevny) [inline mingw-w64-wabt/PKGBUILD:55]:
```suggestion
  cmake --build .
```

**Comment by ognevny** [comment]
please, use a [template](https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-PKGBUILD-templates%2FPKGBUILD.CMake-git.sh) for a PKGBUILD, so all reviews will be applied

**Comment by ArchGuyWu** [comment]
> please, use a [template](https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-PKGBUILD-templates%2FPKGBUILD.CMake-git.sh) for a PKGBUILD, so all reviews will be applied

Ok, I'll switch to use the template.

**Comment by ognevny** [comment]
note: in this case you don't need `pkgver()` function, since you use versioned tag

**Comment by ArchGuyWu** [comment]
> note: in this case you don't need `pkgver()` function, since you use versioned tag

Ok.

**Comment by ArchGuyWu** [comment]
@jeremyd2019 Can you test this package for clangarm64, please? 
It's in the Arch Linux ARM repo, so it should be possible to build it on Windows ARM.

File: mingw-w64-wabt/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  else
+    extra_config+=("-DCMAKE_BUILD_TYPE=Debug")
+  fi
+
+  MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
+    cmake \
+      -GNinja \
+      -DCMAKE_INSTALL_PREFIX="${MINGW_PREFIX}" \
+      "${extra_config[@]}" \
+      -DBUILD_{SHARED,STATIC}_LIBS=ON \
```
Comment (ognevny) [inline mingw-w64-wabt/PKGBUILD:55]:
```suggestion
      -DBUILD_SHARED_LIBS=ON \
```
there is no separate option for static libs

File: mingw-w64-wabt/PKGBUILD [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+pkgver=1.0.36
+pkgrel=1
+pkgdesc="The WebAssembly Binary Toolkit is a suite of tools for WebAssembly (mingw-w64)"
+arch=('any')
+mingw_arch=('mingw64' 'ucrt64' 'clang64' 'clangarm64')
+url="https://github.com/WebAssembly/wabt"
+msys2_references=(
+  'archlinux: wabt'
+)
+license=('APACHE-2.0')
```
Comment (ognevny) [inline mingw-w64-wabt/PKGBUILD:13]:
```suggestion
license=('spdx:Apache-2.0')
```

File: mingw-w64-wabt/PKGBUILD [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs"
+         "${MINGW_PACKAGE_PREFIX}-openssl")
```
Comment (ognevny) [inline mingw-w64-wabt/PKGBUILD:15]: is openssl actually required? I didn't find its dlls in runtime dependencies

Comment (ArchGuyWu) [reply mingw-w64-wabt/PKGBUILD:15]: wasm2c requires libcrypto-3-x64.dll, I saw this in [runtime-dependencies] check.

Comment (ArchGuyWu) [reply mingw-w64-wabt/PKGBUILD:15]:
```
[alarm@ArchRPi ~]$ ldd /usr/bin/wasm2c
        linux-vdso.so.1 (0x0000ffff9d210000)
        libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x0000ffff9ce00000)
        libc.so.6 => /usr/lib/libc.so.6 (0x0000ffff9cc40000)
        /lib/ld-linux-aarch64.so.1 => /usr/lib/ld-linux-aarch64.so.1 (0x0000ffff9d1d3000)
        libm.so.6 => /usr/lib/libm.so.6 (0x0000ffff9cb90000)
        libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x0000ffff9cb50000)
```
But openssl is not required in Arch Linux ARM.

**Comment by ognevny** [comment]
> Can you test this package for clangarm64, please?

@ArchGuyWu first you need to patch https://github.com/WebAssembly/wabt/blob/main/CMakeLists.txt#L117 to be `aarch64|ARM64` instead of `ARM64`

**Comment by ArchGuyWu** [comment]
> > Can you test this package for clangarm64, please?
> 
> @ArchGuyWu first you need to patch https://github.com/WebAssembly/wabt/blob/main/CMakeLists.txt#L117 to be `aarch64|ARM64` instead of `ARM64`

Ok.

File: mingw-w64-wabt/0001-replace-arm64-with-aarch64.patch [hunk 8]
```diff
@@ -0,0 +1,13 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 033544fc..4506fec1 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -114,7 +114,7 @@ if (NOT USE_INTERNAL_SHA256)
+ endif()
+ 
+ if (WIN32)
+-  if(${CMAKE_GENERATOR_PLATFORM} MATCHES "ARM64")
++  if(${CMAKE_GENERATOR_PLATFORM} MATCHES "aarch64")
```
Comment (ognevny) [inline mingw-w64-wabt/0001-replace-arm64-with-aarch64.patch:10]:
```suggestion
+  if(${CMAKE_GENERATOR_PLATFORM} MATCHES "aarch64|ARM64")
```

**Comment by ognevny** [comment]
I think in this case better look on autobuild. I'll disable clangarm64if needed

**Pull request merged**